frontend/src/metabase/visualizations/lib/save-dashboard-pdf.ts

```diff
@@ -0,0 +1,58 @@
+import html2canvas from "html2canvas";
+import jspdf from "jspdf";
+
+import { color } from "metabase/lib/colors";
+
+import { SAVING_DOM_IMAGE_CLASS } from "./save-chart-image";
+
+export const saveDashboardPdf = async (
+  selector: string,
+  dashboardName: string,
+) => {
+  const fileName = `${dashboardName}.pdf`;
+  const node = document.querySelector(selector);
+
+  if (!node || !(node instanceof HTMLElement)) {
+    console.warn("No node found for selector", selector);
+    return;
+  }
+
+  const canvas = await html2canvas(node, {
+    useCORS: true,
+    onclone: (doc: Document, node: HTMLElement) => {
+      node.classList.add(SAVING_DOM_IMAGE_CLASS);
+      node.style.paddingBottom = "20px";
+
+      //This needs to be done manually due to how html2canvas treats styles in SVGs
+      const textNodes = doc.querySelectorAll(
+        ".Card text, .Card tspan",
```

Comment:
Is it possible to similarly move out of here this `textNodes` logic and the `node.style.paddingBottom = "20px";` above? Basically to remove the `onclone` function fully. Besides that looks good!

Reply:
That one is a little harder to do. I'll try taking a closer look at it, but when html2canvas does it's cloning and comes across svg elements, it ends up applying a style tag to the elements with the whole computed style. I didn't have any luck overwriting those with a stylesheet, so I had to modify them this way (it also wasn't properly picking up the lato font, so this was meant to tide us over).

Also, changing the border styles outside of the onClone will cause the screen to change while the canvas is being generated. Doing it in here gives better UX